src/lib.rs

```diff
@@ -4425,6 +4500,69 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_ref_from_mut_from() {
+        // Test `FromBytes::{ref_from, mut_from}{,_prefix,_suffix}` success cases
+        // Exhaustive coverage for these methods is covered by the `Ref` tests above,
+        // which these helper methods defer to.
+
+        let mut buf =
+            Align::<[u8; 16], AU64>::new([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]);
```

Comment:
Can you make this 24 bytes long to avoid bugs like the one discovered in https://github.com/google/zerocopy/pull/520?

Reply:
It's not clear to me how making it 24 bytes would improve anything with respect to the bug described. `suffix` acting the same as `prefix` would be caught by these tests as written - which they actually did catch during writing because I made a copy paste error.

Reply:
Sorry, I should have linked to #506 and #511; #520 was just me backporting to 0.6.x. The bug was in part that we used the wrong value as the location to split the byte slice in two. When the length of the object is exactly half the length of the byte slice, then the length of the object and the location to split at are identical, and so it can hide the bug.

Reply:
Again, I don't see how changing to 24 bytes improves the test as written. The two improvements described:

- Using non-zero values, catching an incorrect position
- Using a non-half amount of the full buffer, catching an incorrect length

are both implemented already. `<[u8; 9]>::ref_from_suffix` and `<[u8; 9]>::ref_from_prefix` both check the non-zero values of the buffer they are in, and do so with a non-half amount of the full buffer.

Reply:
Oh gotcha, I didn't catch the `[u8; 9]`. Maybe leave a comment on those calling them out and explaining why the array length is important? I agree that this covers the bug I was concerned about.

Reply:
Done in #559